vllm/platforms/cpu.py

```diff
@@ -147,11 +148,21 @@ def get_device_total_memory(cls, device_id: int = 0) -> int:
         from vllm.utils.mem_constants import GiB_bytes
 
         kv_cache_space = envs.VLLM_CPU_KVCACHE_SPACE
+        node_dir = "/sys/devices/system/node"
         if kv_cache_space is None:
-            kv_cache_space = 4 * GiB_bytes  # type: ignore
+            nodes = (
+                [d for d in os.listdir(node_dir) if d.startswith("node")]
+                if os.path.exists(node_dir)
+                else []
+            )
+            num_numa_nodes = len(nodes) or 1
+            free_cpu_memory = psutil.virtual_memory().available // num_numa_nodes
```

Comment:
Maybe we should use ```psutil.virtual_memory().total``` to get determinstic size. A test case failed due to changed available size.
https://buildkite.com/vllm/ci/builds/41717#019ae318-4a1b-47c9-8e64-060aacccdaff